src/sssom/writers.py

```diff
@@ -204,6 +218,33 @@ def write_owl(
 # Converters convert a mappingsetdataframe to an object of the supportes types (json, pandas dataframe)
 
 
+def _hydrate_axioms(
+    graph: rdflib.Graph,
+    *,
+    add_negative: bool = True,
+    add_no_term_found: bool = True,
+) -> None:
+    for axiom in graph.subjects(RDF.type, OWL.Axiom):
+        for p in graph.objects(subject=axiom, predicate=OWL.annotatedProperty):
+            for s in graph.objects(subject=axiom, predicate=OWL.annotatedSource):
+                for o in graph.objects(subject=axiom, predicate=OWL.annotatedTarget):
+                    if not add_negative and _is_negated(graph, axiom):
+                        continue
+                    if not add_no_term_found and _is_no_term_found(s, o):
+                        continue
+                    graph.add((s, p, o))
+
+
+def _is_no_term_found(s: rdflib.Node, o: rdflib.Node) -> bool:
+    return s == NO_TERM_FOUND_URI or o == NO_TERM_FOUND_URI
+
+
+def _is_negated(graph: rdflib.Graph, axiom: rdflib.Node) -> bool:
+    return any(
+        obj == rdflib.Literal("NOT") for obj in graph.objects(subject=axiom, predicate=NEGATED)
```

Comment:
This works for now, but my understanding of the way the LinkML RDF graph “dumper“ works is that, once we will have updated SSSOM-Py to use the latest version of the schema, the `Not` value of the `predicate_modifier` slot will automatically be rendered as a `https://w3id.org/sssom/NegatedPredicate` resource, instead of a `Not` string literal (since in the update schema the `Not` value has a `meaning` property set to `sssom:NegatedPredicate`).

So when we do update to the next version of `sssom`, this function will need to be amended accordingly.

Maybe you’d want to future-proof it right now by looking for _either_ a `Not` string literal _or_ a `sssom:NegatedPredicate` resource?